src/coord/src/coord.rs

```diff
@@ -4130,7 +4191,11 @@ impl Coordinator {
         // message so we can persist a record and its future retraction
         // atomically. Otherwise, we may end up with permanent orphans if a
         // restart/crash happens at the wrong time.
-        let timestamp_base = self.get_local_write_ts();
+        let timestamp_base = match self.get_local_write_ts() {
+            Ok(ts) => ts,
+            // If no write timestamp can be produced, drop this set of updates.
```

Comment:
If this occurs for a catalog-representing system table, there's nothing that will ever re-populate the table with the item. This seems like a really bad state to be in; could we plumb a toggle here for enabling or disabling this behavior?

Reply:
I do not know! @antifuchs said it was fine to drop these. Maybe he knows why this is safe.

Reply:
Huh! I don't know if I'm remembering what I said about this - when would that have been? In any case, if Sean says there's a potential problem, I'd trust him more than me! (-:

Reply:
[this call](https://github.com/MaterializeInc/materialize/blob/main/src/coord/src/coord.rs#L607) after `bootstrap` winds up calling this code path and if e.g. this is where we hit the failure, we will be sad. It might be fine for metrics scraping, though--idk.

Reply:
My recollection was that it was safe to drop metrics if we aren't able to get a timestamp. But Sean, yes, you're right here because non-metrics stuff also calls this function, and those cannot be dropped.

Reply:
Oh - yeah, for mz_metrics, sure! That's bound to be lossy. But this code path is used for general insertion of things in tables, so the constraints aren't necessarily those of the metrics table. Maybe that's what Sean is calling out?

Reply:
Yeah, exactly

Reply:
Re-jiggered to now have 2 write_ts functions, one that can fail and one that forcibly moves things ahead. I...do not like this but don't have a better idea and am like 99% sure now that it has some fatal flaw that will prevent merging. Maybe? I really want to get the table code ready for whatever changes are coming from persist/storage/platform and thought this was the way, but maybe it is not.

Reply:
Ooh! What if the timestamp oracle stays infallible, but coord instead is incharge of saying "hey, looks like the timestamp you gave me is ahead of now(), so i'm just going to wait", and the oracle doesn't know or care about that?

Reply:
That feels like oracular. Like, if you're second-guessing your oracle's information, that feels like it ought to be called by some other name. I'm not opposed to the code doing that, but the naming becomes less intuitive at that point.

Reply:
Another take on that: the oracle tells us about a timestamp that has some properties, but those properties do not include "it's in advance of now()", because the oracle doesn't care about now, as it is timeline-independent. Instead, consumers of the oracle are in charge of acting on that information by, say, waiting in reality until the timestamp syncs with now().